compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -26,13 +26,13 @@ use std::rc::Rc;
 
 pub use self::select::{EvaluationCache, EvaluationResult, OverflowError, SelectionCache};
 
-pub type ChalkCanonicalGoal<'tcx> = Canonical<'tcx, ChalkEnvironmentAndGoal<'tcx>>;
+pub type CanonicalChalkGoal<'tcx> = Canonical<'tcx, ChalkGoal<'tcx>>;
```

Comment:
Renamed for consistency with the goals in `rustc_middle::traits::query` such as `CanonicalTypeOpEqGoal`.

Reply:
Rename is fine, but should still be `ChalkEnvironmentAndGoal`